homeassistant/components/smartthings/config_flow.py

```diff
@@ -113,41 +124,34 @@ def __init__(self):
 
     async def async_step_wait_install(self, user_input=None):
         """Wait for SmartApp installation."""
-        from pysmartthings import InstalledAppStatus
-
         errors = {}
         if user_input is None:
             return self._show_step_wait_install(errors)
 
         # Find installed apps that were authorized
-        installed_apps = [app for app in await self.api.installed_apps(
-            installed_app_status=InstalledAppStatus.AUTHORIZED)
-                          if app.app_id == self.app_id]
+        installed_apps = self.hass.data[DOMAIN][CONF_INSTALLED_APPS].copy()
         if not installed_apps:
             errors['base'] = 'app_not_installed'
             return self._show_step_wait_install(errors)
+        self.hass.data[DOMAIN][CONF_INSTALLED_APPS].clear()
+
+        # Enrich the data
+        for installed_app in installed_apps:
+            installed_app[CONF_APP_ID] = self.app_id
+            installed_app[CONF_ACCESS_TOKEN] = self.access_token
+            installed_app[CONF_OAUTH_CLIENT_ID] = self.oauth_client_id
```

Comment:
See https://github.com/home-assistant/architecture/issues/139 about storing client id and secret.

Reply:
I didn't see an outcome/recommendation in that topic at this time.  In this case, these are obtained programmatically, so not something the user could enter/see.  I can move them to the smartthings storage file, but it adds unnecessary complexity, as it would have to be keyed to the app/config entry anyway.  Do you have a recommendation on what you'd like to see?

Reply:
Right, we're not asking the user to enter them in the config flow just storing them in the config entry. Then I think it's ok. @balloob?

Reply:
Yeah, its ok if user is not entering them.